lightning/src/ln/channelmanager.rs

```diff
@@ -2293,8 +2295,14 @@ where
 			for (_cp_id, peer_state_mutex) in per_peer_state.iter() {
 				let mut peer_state_lock = peer_state_mutex.lock().unwrap();
 				let peer_state = &mut *peer_state_lock;
-				// Only `Channels` in the channel_by_id map can be considered funded.
-				for (_channel_id, channel) in peer_state.channel_by_id.iter().filter(f) {
+				for (_channel_id, channel) in peer_state.channel_by_id.iter()
+					.filter_map(|(chan_id, phase)| match phase {
+						// Only `Channels` in the `ChannelPhase::Funded` phase can be considered funded.
+						ChannelPhase::Funded(chan) => Some((chan_id, chan)),
+						_ => None,
+					})
+					.filter(f)
+				{
 					let details = ChannelDetails::from_channel_context(&channel.context, best_block_height,
 						peer_state.latest_features.clone(), &self.fee_estimator);
```

Comment:
nit: could probably just add this as another `.map` and then `res.extend` the whole iterator (and similarly in `list_channels`)